### ISSUE TITLE
Update sort-comp for React 16.3+

### ISSUE THREAD

**Issue by webOS101** [yannickcr/eslint-plugin-react#1767]
The `sort-comp` rule will need to be updated to include the new lifecycle methods (while also keeping the older ones, optionally?).  These include:

```
getDerivedStateFromProps
getSnapshotBeforeUpdate
UNSAFE_componentWillMount
UNSAFE_componentWillReceiveProps
UNSAFE_componentWillUpdate
```

**Comment by otakustay** [comment]
Also, `componentDidCatch` is not listed in current `sort-comp` rule

**Comment by joe-denea** [comment]
I made an attempt to add this feature with #1771 

Would love to hear any feedback/suggestions

**Comment by ljharb** [comment]
Closed in #1771.
